src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+interface SortSpecs {
+  sortBy: SortType | null;
+  reversed: boolean;
+}
+
+export const App = () => {
+  const getVisibleGoods = (
+    goods: string[],
+    { sortBy, reversed }: SortSpecs,
+  ): string[] => {
+    const modified = [...goods];
+
+    if (sortBy === SortType.Name) {
+      modified.sort((g1, g2) => {
+        return g1.localeCompare(g2);
+      });
+    }
+
+    if (sortBy === SortType.Length) {
+      modified.sort((g1, g2) => {
+        return g1.length - g2.length;
+      });
+    }
+
+    if (reversed) {
+      modified.reverse();
+    }
+
+    return modified;
+  };
+
+  const [sortBy, setSortBy] = useState<SortType | null>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const prepareGoods = getVisibleGoods(goodsFromServer, { sortBy, reversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SortType.Name,
+          })}
+          onClick={() =>
+            setSortBy(sortBy === SortType.Name ? null : SortType.Name)
+          }
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SortType.Length,
+          })}
+          onClick={() =>
+            setSortBy(sortBy === SortType.Length ? null : SortType.Length)
+          }
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => {
+            setReversed(() => !reversed);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortBy || reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here